sklearn/utils/validation.py

```diff
@@ -295,8 +295,10 @@ def check_array(array, accept_sparse=None, dtype="numeric", order=None,
         If dtype is a list of types, conversion on the first type is only
         performed if the dtype of the input is not in the list.
 
-    order : 'F', 'C' or None (default=None)
-        Whether an array will be forced to be fortran or c-style.
+    order : 'F', 'C' or 'A' (default='A')
```

Comment:
I'm remember you telling that this is some NumPy bug, but this is really odd!

1. Which part of the code in this PR does this break if order is set to None?
2. Does enforcing the order of `check_array(X) = X`, break code anywhere else?